### ISSUE TITLE
Use projext for bundling the packages

### ISSUE THREAD

**Issue by joelalejandro** [xethya/framework#8]
It'd be nice to start using [projext](https://github.com/homer0/projext) to simplify the framework's bundling process.

@homer0, what'd be the best way to implement your bundler? This monorepo is entirely built with TypeScript, and currently using Rollup to bundle the packages. The idea is to output each package as an importable library or as an IIFE (see [this file](https://github.com/xethya/framework/tree/master/packages/dice/rollup.config.json) for reference).

**Comment by homer0** [comment]
TL;DR: It can't be used for now, I found a tiny bug and three limitations: two were already on my roadmap, the other it from outside projext.

---

Ok, so I cloned the repo and started to play around to see how complicated it would be to run it using projext.

## Overview

This is the first use case that is truly different from the type of things projext is usually used for:

- It's not an app.
- It's a monorepo.
- It makes "heavy" use of TypeScript.

I expected things to go **so** wrong, but it wasn't as bad as I thought :D.

My recommendation is to close the issue as "projext is not ready" (:'(, but it's the truth), and I'll keep you updated whenever I add the features/fixes Xethya will need in order to use it.

There's a list below with all the things I found:

## Issues

### Monorepo

While I'm very familiar with the concept and I know the tools needed for it, I've never actually used one, so as you may imagine, I never tested projext on a monorepo.

I know the setup is simple, but I didn't have enough time to read about it and then do the tests, so next time we get together, you can help me test that part; I have a few theories about some things that may fail due to the execution directory being different from the package directory... but they're theories, I can't be sure yet :P.

So, I took the `dice` package, moved to another directory (outside the monorepo) and did the rest of the tests.

### Names with scopes

This is the tiny bug I mentioned above; one of the advantages of projext's "zero configuration" is that you don't even need to have an HTML file to import your JS when you start, it will create a temporarily file for you that you can later replace.

In the case the target is a library, like in this case, it only creates the HTML while on development mode, so you can test the library on the browser... that's when I saw the error: projext was trying to create `.../.tmp/@xethya/dice.index.html` because the format is `[target-name].index.html`.

This is really simple to fix and I'll include it on the next minor path; While making the tests, I just added a `.replace(/\//g, '-')` and the issue was gone :P.

### Multiple outputs

You Rollup configurations have outputs for ES modules, commonjs and IIFE, and projext doesn't have support for it yet. The truth is that every time I add a feature, I add it both for the webpack and Rollup plugin, because one of the objectives of projext is that you can choose between any of those two and you won't have to learn a new config.

This is one of the things that was already on my roadmap (along with multiple entries), the thing is that for webpack is not so simple: you can have multiple outputs if you have multiple entries, otherwise, you need to export multiple configs.

Since I have a real use case now, I can give it more priority and probably include it on the next feature release.

### 'const' enums are not supported

The TypeScript support of projext comes from Babel, and it seems like this is a limitation of the plugin. I found [this issue](https://github.com/babel/babel/issues/8741) and it seems like there's a [plugin](https://www.npmjs.com/package/babel-plugin-const-enum) for it, or you can simply remove the `const`.

I would love for you to give me some input here as you have more experience with TypeScript; for purpose of the test, I removed the only `const` you have on the package and I was able to build it.

**Comment by joelalejandro** [comment]
This is an excellent sit-rep! I'll just put this issue on hold until proper support is there. I have other plenty of stuff to build, lol.

About const enums, [this](https://stackoverflow.com/a/28818850) is a very good explanation of why I use that one instead of the regular, non-const `enum` _(TL;DR: `const enum` gets transpiled, `enum` is restricted to TypeScript land)_.

Happy to help out to expand projext for this use case! :)
